### PR TITLE
[feat] allow returning redirect from `handleError`

### LINKED ISSUE #2824: Have a way to render an error page when there's an exception in a hook (e.g. handle)
### Describe the problem

When an exception occurs in handle you get the exception and stack trace in the response to the user.

### Describe the proposed solution

It would be great if there was a way to render an error page.

### Alternatives considered

_No response_

### Importance

nice to have

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by nhe23**
Fixes #2824 by allowing the user to optionally return a redirect value for a custom error page. This way an error page can be rendered even if a hook throws an exception.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/test/apps/basics/src/hooks.js [hunk 1]
```diff
@@ -18,6 +18,20 @@ export const handleError = ({ request, error }) => {
 		: {};
 	errors[request.url.pathname] = error.stack || error.message;
 	fs.writeFileSync('test/errors.json', JSON.stringify(errors));
+	console.error(error.message);
```
Comment (benmccann) [inline packages/kit/test/apps/basics/src/hooks.js:21]: I'm not sure we'd want to be logging to the console even in tests

File: packages/kit/test/apps/basics/src/hooks.js [hunk 2]
```diff
@@ -18,6 +18,20 @@ export const handleError = ({ request, error }) => {
...
+	if (error.frame) {
+		console.error(error.frame);
+	}
+	if (error.stack) {
+		console.error(error.stack);
+	}
+
+	if (error.message === '"Testing hook exception"') {
```
Comment (benmccann) [inline packages/kit/test/apps/basics/src/hooks.js:29]: I wonder if the double quotes are needed here?

File: documentation/docs/04-hooks.md [hunk 3]
```diff
@@ -96,6 +100,12 @@ export interface HandleError<Locals = Record<string, any>, Body = unknown> {
 export async function handleError({ error, request }) {
 	// example integration with https://sentry.io/
 	Sentry.captureException(error, { request });
+
+	//Optionally return a status and redirect destination
```
Comment (benmccann) [inline documentation/docs/04-hooks.md:104]: we always put a space after `//`
```suggestion
	// Optionally return a status and redirect destination
```

File: packages/kit/test/apps/basics/src/hooks.js (hunk 2 above)
Comment (nhe23) [reply packages/kit/test/apps/basics/src/hooks.js:29]: They are needed. Somehow the error that gets thrown in the handle hook is wrapped in double quotes. As my changes did not cause this I thought it was the wanted behaviour and did not look into why it is wrapped in double quotes.

File: packages/kit/test/apps/basics/src/hooks.js (hunk 1 above)
Comment (nhe23) [reply packages/kit/test/apps/basics/src/hooks.js:21]: Prior to the recent switch to playwright there was no handleErrors hook in the tests. That's why I mimicked the default implementation (which is logging errors to console.error) as some tests were depending on the output to be exactly this way. But with the recent changes this is not needed anymore and I'll remove the console.errors

**Comment by nhe23** [comment]
I have no idea why the `pnpm` check command in the ci fails. It works on my machine and the error saying `Property 'started' does not exist on type 'PlaywrightTestArgs...` does not make any sense to me, as im using 'started' in exactly the same way as other tests.

File: packages/kit/test/apps/basics/test/test.js [hunk 4]
```diff
@@ -620,6 +619,21 @@ test.describe.parallel('Errors', () => {
 			expect(await page.innerHTML('h1')).toBe('401');
 		}
 	});
+
+	test('custom error page is rendered if handle hook throws an error', async ({
+		baseURL,
+		page,
+		javaScriptEnabled,
+		started
```
Comment (benmccann) [inline packages/kit/test/apps/basics/test/test.js:627]: if you rebase against `master` there is no `started` anymore. You should rebase and then delete this, the `await started()`, and the `waitForTimeout` lines

Comment (nhe23) [reply packages/kit/test/apps/basics/test/test.js]: Thank you for the hint. I thought that I already had rebased, but that must have been another branch I was working on ... 🤦

File: packages/kit/test/apps/basics/src/hooks.js (hunk 1 above)
Comment (nhe23) [reply packages/kit/test/apps/basics/src/hooks.js]: Removed console.error from test

File: packages/kit/test/apps/basics/test/test.js (hunk 4 above)
Comment (nhe23) [reply packages/kit/test/apps/basics/test/test.js]: Now it works. The failing test on windows is also failing in other PRs ... so this is probably not the fault of my changes, right?

**Comment by Rich-Harris** [comment]
closing per https://github.com/sveltejs/kit/pull/3235#issuecomment-1007798838

**Pull request closed without merging**